tool/tsh/db.go

```diff
@@ -109,6 +113,95 @@ func onListDatabases(cf *CLIConf) error {
 	return trace.Wrap(showDatabases(cf.SiteName, databases, activeDatabases, roleSet, cf.Format, cf.Verbose))
 }
 
+type databaseListing struct {
+	Proxy    string           `json:"proxy"`
+	Cluster  string           `json:"cluster"`
+	roleSet  services.RoleSet `json:"-"`
+	Database types.Database   `json:"database"`
+}
+
+func listDatabasesAllClusters(cf *CLIConf) error {
+	profile, profiles, err := client.Status(cf.HomePath, "")
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if profile != nil {
+		profiles = append(profiles, profile)
+	}
+
+	dbListings := make([]databaseListing, 0)
+	for _, p := range profiles {
+		cf.Proxy = p.ProxyURL.Host
+		tc, err := makeClient(cf, true)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		result, err := tc.ListDatabasesAllClusters(cf.Context, nil /* custom filter */)
+		// Don't make the user re-login to every proxy.
+		if client.IsExpiredCredentialError(err) {
+			fmt.Fprintf(os.Stderr, "Credentials expired for proxy %q, skipping...\n", cf.Proxy)
+			continue
+		}
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		proxy, err := tc.ConnectToProxy(cf.Context)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		for clusterName, databases := range result {
+			cluster, err := proxy.ConnectToCluster(cf.Context, clusterName, false)
+			if err != nil {
+				return trace.Wrap(err)
+			}
+			roleSet, err := services.FetchRoles(p.Roles, cluster, p.Traits)
+			if err != nil {
+				return trace.Wrap(err)
+			}
+			for _, database := range databases {
+				dbListings = append(dbListings, databaseListing{
+					Proxy:    cf.Proxy,
+					Cluster:  clusterName,
+					roleSet:  roleSet,
+					Database: database,
+				})
+			}
+		}
+	}
+
+	sort.Slice(dbListings, func(i, j int) bool {
```

Comment:
Nit: I think it would be cleaner (and more reusable) if you made `databaseListing` satisfy [sort.Interface](https://pkg.go.dev/sort#Interface) and then just `sort.Sort(dbListings)` here.